src/Common/FileCache.cpp

```diff
@@ -473,40 +641,42 @@ bool LRUFileCache::tryReserve(const Key & key, size_t offset, size_t size, std::
     else
     {
         size_t removed_size = 0;
-        size_t queue_size = queue.getElementsNum(cache_lock);
+        size_t queue_size = main_priority->getElementsNum(cache_lock);
 
         auto * cell_for_reserve = getCell(key, offset, cache_lock);
 
-        std::vector<IFileCache::LRUQueue::Iterator> ghost;
+        std::vector<std::tuple<Key, size_t, size_t>> ghost;
```

Comment:
Better to use struct instead of tuple here -- to make it more clear what the values are.